test/Microsoft.AspNet.Mvc.TagHelpers.Test/EnvironmentTagHelperTest.cs

```diff
@@ -0,0 +1,149 @@
+        }
+
+        [Theory]
+        [InlineData("NotDevelopment", "Development")]
+        [InlineData("NOTDEVELOPMENT", "Development")]
+        [InlineData("NotDevelopment,AlsoNotDevelopment", "Development")]
+        [InlineData("Doesn'tMatchAtAll", "Development")]
+        [InlineData("Development and a space", "Development")]
+        [InlineData("Development and a space,SomethingElse", "Development")]
+        public void DoesNotShowContentWhenCurrentEnvironmentIsNotSpecified(string namesAttribute, string environmentName)
+        {
+            // Arrange
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object> { { "names", namesAttribute } },
+                content: content);
+            var output = MakeTagHelperOutput("environment");
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = environmentName;
+
+            // Act
+            var helper = new EnvironmentTagHelper
+            {
+                HostingEnvironment = hostingEnvironment.Object,
+                Names = namesAttribute
+            };
+            helper.Process(context, output);
+
+            // Assert
+            Assert.Null(output.TagName);
+            Assert.Null(output.PreContent);
+            Assert.Null(output.Content);
+            Assert.Null(output.PostContent);
+            Assert.True(output.ContentSet);
+        }
+
+        private void ShouldShowContent(string namesAttribute, string environmentName)
+        {
+            // Arrange
+            var content = "content";
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object> { { "names", namesAttribute } },
+                content: content);
+            var output = MakeTagHelperOutput("environment");
+            var hostingEnvironment = new Mock<IHostingEnvironment>();
+            hostingEnvironment.SetupProperty(h => h.EnvironmentName);
+            hostingEnvironment.Object.EnvironmentName = environmentName;
+
+            // Act
+            var helper = new EnvironmentTagHelper
+            {
+                HostingEnvironment = hostingEnvironment.Object,
+                Names = namesAttribute
+            };
+            helper.Process(context, output);
+
+            // Assert
+            Assert.Null(output.TagName);
+            Assert.False(output.ContentSet);
+        }
+
+        private TagHelperContext MakeTagHelperContext(IDictionary<string, object> attributes = null, string content = null)
+        {
+            if (attributes == null)
+            {
+                attributes = new Dictionary<string, object>();
+            }
+
+            return new TagHelperContext(attributes, Guid.NewGuid().ToString("N"), () => Task.FromResult(content));
```

Comment:
Can simplify/remove above if statement if you want to do: `attributes ?? new Dictionary<string, object>()`